### ISSUE TITLE
Using --writable-tmpfs fails run with fuse-overlayfs

### ISSUE THREAD

**Issue by afbjorklund** [apptainer/apptainer#796]
### Version of Apptainer

What version of Apptainer (or Singularity) are you using? Run

`apptainer --version` (or `singularity --version`).

```
apptainer version 1.1.2
```

### Expected behavior

What did you expect to see when you do...?

```
$ apptainer run --writable-tmpfs docker://alpine
INFO:    Using cached SIF image
Apptainer> 
```

### Actual behavior

What actually happened? Why was it incorrect?

```
$ apptainer run --writable-tmpfs docker://alpine
INFO:    Using cached SIF image
FATAL:   container creation failed: mount /var/tmp->/var/tmp error: while mounting /var/tmp: could not mount /var/tmp: operation not supported
```

```
DEBUG   [U=1000,P=31972]   mountGeneric()                Mounting /tmp to /var/apptainer/mnt/session/final/tmp
DEBUG   [U=1000,P=31972]   mountGeneric()                Remounting /var/apptainer/mnt/session/final/tmp
DEBUG   [U=1000,P=31972]   mountGeneric()                Mounting /var/tmp to /var/apptainer/mnt/session/final/var/tmp
DEBUG   [U=1000,P=31972]   umount()                      Umount /var/apptainer/mnt/session/final
```

### Steps to reproduce this behavior

How can others reproduce this issue/problem?

`--writable-tmpfs`

Host mounts:

```
/dev/vda5 on / type btrfs (rw,relatime,seclabel,compress=zstd:1,space_cache=v2,subvolid=256,subvol=/root)
tmpfs on /tmp type tmpfs (rw,nosuid,nodev,seclabel,nr_inodes=1048576,inode64)
```

No issues seen when running on Ubuntu with ext4.

```
/dev/nvme0n1p2 on / type ext4 (rw,relatime,errors=remount-ro)
```

Side note:
Ironically installing `apptainer-suid` also fails, but for a different reason...

`FATAL:   container creation failed: mount /proc/self/fd/3->/var/apptainer/mnt/session/rootfs error: while mounting image /proc/self/fd/3: squashfs filesystem seems not enabled and/or supported by your kernel`

The solution to that problem is: `dnf install -y kernel-modules-$(uname -r)`

### What OS/distro are you running

`PRETTY_NAME="Fedora Linux 36 (Cloud Edition)"`

### How did you install Apptainer

Write here how you installed Apptainer (or Singularity). Eg. RPM, source.

`dnf -y install apptainer`

apptainer-1.1.2-1.fc36.x86_64

**Comment by afbjorklund** [comment]
This fuse-overlay version works:

fuse-overlayfs-1.3.0-1.fc33.x86_64

This fuse-overlay version fails:

fuse-overlayfs-1.9-6.fc36.x86_64

----

Seems to be consistent with this issue:
* https://github.com/containers/fuse-overlayfs/issues/232

**Comment by afbjorklund** [comment]
It seems to be related to the POSIX ACL changes:

```
[pid  3172] lgetxattr("/var/apptainer/mnt/session/rootfs/var", "system.posix_acl_access", 0x55bb250252f0, 4096) = -1 EOPNOTSUPP (Operation not supported)
```

It now uses the full path, which forces a lgetxattr ?

https://github.com/containers/fuse-overlayfs/commit/4ad759b35a054b6010b348755921977a9dcf6e0e

**Comment by afbjorklund** [comment]
Seems like podman happily ignores `EOPNOTSUPP`, when listing the xattr.

**Comment by DrDaveD** [comment]
This can be reproduced also on rocky8, which has fuse-overlayfs-1.9.

**Comment by afbjorklund** [comment]
> This can be reproduced also on rocky8, which has fuse-overlayfs-1.9.

It uses an older Fedora, but otherwise it should be same issue - I think ?

The CentOS 8-Stream has: `fuse-overlayfs-1.6-1.module+el8.5.0+11957+18363336`

https://git.centos.org/rpms/fuse-overlayfs/history/SPECS/fuse-overlayfs.spec?identifier=c8s-stream-rhel8

**Comment by DrDaveD** [comment]
Fixed in #823.

It doesn't make sense that CentOS-8 Stream would have an older version than Rocky8.  Your git.centos.org link shows 1.9 for el8.6.  1.6 was in el8.4.

**Comment by afbjorklund** [comment]
> Fixed in #823.
> 
> It doesn't make sense that CentOS-8 Stream would have an older version than Rocky8. Your git.centos.org link shows 1.9 for el8.6. 1.6 was in el8.4.

I meant it "used to have", it was a way to test if the introduction of the ACL caused the regression - thanks for the fix

**Comment by DrDaveD** [comment]
Note that ubuntu 22.04 jammy is still subject to this problem.  fuse-overlayfs there is based on version 1.7, so using fakeroot with overlayfs (with apptainer 1.1.6) results in an error
```
FATAL:   container creation failed: mount hook function failure: mount /var/tmp->/var/tmp error: while mounting /var/tmp: could not mount /var/tmp: operation not supported
```

A workaround is to update to fuse-overlayfs-1.9 from 22.10 kinetic, for example by doing this as root:
```
echo "deb http://archive.ubuntu.com/ubuntu kinetic universe" >/etc/apt/sources.list.d/kinetic-universe.list
cat >/etc/apt/preferences.d/kinetic.pref <<xEOFx
Package: *
Pin: release a=kinetic
Pin-Priority: 50
xEOFx
apt-get update
apt-get install -y -t kinetic fuse-overlayfs
```
